hopr-socks/Cargo.toml

```diff
@@ -0,0 +1,23 @@
+[package]
+name = "hopr-socks"
+version = "0.1.0"
+authors = ["HOPR Association <tech@hoprnet.org>"]
+edition = "2021"
+description = ""
+homepage = "https://hoprnet.org/"
+repository = "https://github.com/hoprnet/hoprnet"
+license = "GPL-3.0-only"
+
+[dependencies]
+fast-socks5 = "0.9.6"
+env_logger = "0.10.2"
+structopt = "0.3"
```

Comment:
`structopt` is now obsolete (see https://github.com/texitoi/structopt?tab=readme-ov-file#maintenance).

We can use `clap` right away, and use the `workspace` version of `clap`.

Reply:
Addressed.